src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp

```diff
@@ -53,9 +53,20 @@ void FleetApiClient::setDeviceIdentification(const fleet_protocol::cxx::DeviceID
 }
 
 
-std::vector<std::shared_ptr<model::Car>> FleetApiClient::getCars() {
-	auto carsRequest = carApi_->availableCars();
-	return carsRequest.get();
+std::vector<std::shared_ptr<model::Car>> FleetApiClient::getCars(std::optional<int64_t> since, std::optional<bool> wait) {
+	auto carsRequest = carApi_->availableCars(wait.value_or(false), since.value_or(0));
+	std::vector<std::shared_ptr<model::Car>> cars;
+
+	try {
+		cars = carsRequest.get();
+	} catch(std::exception &e) {
```

Comment:
Why is this catch empty?

Reply:
Any http response other than 2XX causes an exception, which would produce error messages in logs.
The try catch blocks are there to prevent exception spam (if the functions are being called constantly) whenever the http api is unavailable or returns 404 for example.